### ISSUE TITLE
BUG: `getAllByIndexSync` return wrong id

### ISSUE THREAD

**Issue by jtzell** [isar/isar#207]
**Steps to Reproduce**
**Model**
```
@Collection
class Model {
  int? id;
  @Id()
  int? offlineId;
  @Index(unique: true, type: IndexType.value, caseSensitive: true)
  String guid;

  Model({required this.guid, this.id, this.offlineId});
}
```
Use `putSync` to save model, and retrieved it back by `guid`.
```
final model = Model(guid: Uuid().v4(), id: 1)
_isar.writeTxnSync((isar) {
  final txnCol = isar.getCollection<Model>('Model');
  txnCollection.putSync(model, replaceOnConflict: true);
  final savedModel = txnCollection.getAllByIndexSync('guid', [
    [model.guid]
  ]).first!;
  assert(null != savedModel.offlineId &&
            savedModel.offlineId == model.offlineId);
}
```
The above code will throw assertion exception, since `offineId` return by `getAllByIndexSync` is not same as `model.offlineId` assigned to the model by `putSync`.

**Version**
 - Platform: iOS, Android, Mac, Windows, Linux, Web
 - Flutter version: 2.8.1
 - Isar version: 2.0.0
 - Isar Generator version: 2.0.0

**Comment by leisim** [comment]
Thanks for the report! Does it work with `getAllByIndex()` (the async version of this method)?

**Comment by jtzell** [comment]
> Thanks for the report! Does it work with `getAllByIndex()` (the async version of this method)?

Yes, `getAllByIndex()` also have the same problem.

Moreover, I'd like to note that the field `guid` is annotated with `@Index(unique: true, type: IndexType.value, caseSensitive: true)`. I forgot the mention that in the previous comment (and just edit it).

**Comment by leisim** [comment]
Fixed in `2.1.0` 🙌
